### ISSUE TITLE
0.2.10 node hard fork with block explorer

### ISSUE THREAD

**Issue by lcgogo** [maticnetwork/bor#253]
My 0.2.10 node and 0.2.9 node syncs diff hash blocks with https://polygonscan.com/

```
INFO [12-05|08:01:30.113] Imported new chain segment               blocks=1  txs=100  mgas=21.290  elapsed=151.819ms   mgasps=140.232 number=22,156,804 hash=424a6f..a6c21e dirty=252.81MiB
INFO [12-05|08:01:30.115] Unindexed transactions                   blocks=1  txs=78   tail=19,806,805 elapsed=1.497ms
INFO [12-05|08:01:38.728] Imported new chain segment               blocks=1  txs=222  mgas=21.275  elapsed=203.355ms   mgasps=104.620 number=22,156,805 hash=b5ed7b..5c4dc5 dirty=252.49MiB
INFO [12-05|08:01:38.730] Unindexed transactions                   blocks=1  txs=151  tail=19,806,806 elapsed=2.096ms
INFO [12-05|08:01:46.590] Imported new chain segment               blocks=1  txs=140  mgas=21.264  elapsed=223.498ms   mgasps=95.143  number=22,156,806 hash=bfdd4f..5b9084 dirty=252.41MiB
INFO [12-05|08:01:46.592] Unindexed transactions                   blocks=1  txs=107  tail=19,806,807 elapsed=1.370ms
INFO [12-05|08:01:54.844] Imported new chain segment               blocks=1  txs=141  mgas=21.244  elapsed=291.239ms   mgasps=72.942  number=22,156,807 hash=8bec6e..35476f dirty=252.69MiB
```

**Comment by lcgogo** [comment]
Think the hard fork starts from about block 22156666.

**Comment by masayil** [comment]
My 0.2.12 node and 0.2.10 nodes have the same problem.
The test nodes work well

**Comment by vae520283995** [comment]
same issue

**Comment by masayil** [comment]
> same issue

https://forum.polygon.technology/t/v0-2-12-beta3-upgrade-for-bor-mainnet/360
Pls check this website

**Comment by lcgogo** [comment]
> > same issue
> 
> https://forum.polygon.technology/t/v0-2-12-beta3-upgrade-for-bor-mainnet/360 Pls check this website

Thanks, will try.

**Comment by lcgogo** [comment]
> > same issue
> 
> https://forum.polygon.technology/t/v0-2-12-beta3-upgrade-for-bor-mainnet/360 Pls check this website

After both node update to 0.2.12-beta3 and debug.setHead back to block 22156400 which about 200 blocks before hard fork, the node sync for a while.
But now both the node sync stucks at block 22157055 with `Error: invalid merkle root ` like below

```
Error: invalid merkle root (remote: 6285eeb75ee4e86374a83a21d6c5dc0dc78db4edde8d24f3adfe690b7dd2a2ef local: fda9a13d76662405b23e45d9280e063c11b7b2bb74455bd2b49e1cdfab86c43c)
##############################

WARN [12-05|12:01:22.384] Synchronisation failed, dropping peer    peer=22fb22dc72570abb4faa7217319e8fed3c3183a4b46cb949c003c3adb3775585 err="retrieved hash chain is invalid: invalid merkle root (remote: 6285eeb75ee4e86374a83a21d6c5dc0dc78db4edde8d24f3adfe690b7dd2a2ef local: fda9a13d76662405b23e45d9280e063c11b7b2bb74455bd2b49e1cdfab86c43c)"
ARN [12-05|12:01:22.385] Synchronisation failed, retrying         err="peer is unknown or unhealthy"
INFO [12-05|12:01:22.419] Generating state snapshot                root=88aa67..7d0429 in=4567f3..406c48 at=f5cb98..a01998 accounts=13,549,678 slots=129,256,864 storage=10.52GiB   elapsed=48m35.238s  eta=2h10m37.44s
INFO [12-05|12:01:30.427] Generating state snapshot                root=88aa67..7d0429 in=457d29..42876f at=1bb130..c1b93a accounts=13,566,105 slots=129,725,202 storage=10.55GiB   elapsed=48m43.245s  eta=2h10m46.111s
INFO [12-05|12:01:38.435] Generating state snapshot                root=88aa67..7d0429 in=457f51..c11949 at=4a0581..ecfb3e accounts=13,567,761 slots=130,310,538 storage=10.60GiB   elapsed=48m51.253s  eta=2h11m6.297s
ERROR[12-05|12:01:43.750] Error in block freeze operation          err="block receipts missing, can't freeze block 18450232"
```

**Comment by vae520283995** [comment]
Synchronized to the wrong chain after the upgrade
```
NFO [12-05|14:05:53.310] Aborting state snapshot generation       root=536f72..a28822 in=20898f..476185 at=194f17..64ce6e accounts=6,360,999 slots=38,902,253 storage=3.35GiB    elapsed=16m50.497s  eta=1h55m39.98s
INFO [12-05|14:05:53.311] Resuming state snapshot generation       root=4d04b4..b11a33 in=20898f..476185 at=194f17..64ce6e accounts=6,360,999 slots=38,902,253 storage=3.35GiB    elapsed=16m50.497s  eta=1h55m39.98s
INFO [12-05|14:05:53.318] Imported new chain segment               blocks=1    txs=142   mgas=22.398  elapsed=706.526ms   mgasps=31.702  number=22,157,248 hash=6efae6..981eb5 dirty=229.78MiB
INFO [12-05|14:05:53.321] Unindexed transactions                   blocks=1    txs=321   tail=19,807,249 elapsed=3.512ms
INFO [12-05|14:05:57.959] Deep froze chain segment                 blocks=5    elapsed=22.161ms    number=22,067,248 hash=a325a4..6a21d6
INFO [12-05|14:06:01.318] Generating state snapshot                root=4d04b4..b11a33 in=20bd97..0a7b6d at=8d3a42..b825b1 accounts=6,400,932 slots=39,254,200 storage=3.38GiB    elapsed=16m58.504s  eta=1h55m45.225s
INFO [12-05|14:06:09.318] Generating state snapshot                root=4d04b4..b11a33 at=211a0f..b1f322 accounts=6,471,761 slots=39,364,556 storage=3.40GiB    elapsed=17m6.505s   eta=1h55m12.201s
INFO [12-05|14:06:12.421] Aborting state snapshot generation       root=4d04b4..b11a33 at=2147a2..323221 accounts=6,506,874 slots=39,386,076 storage=3.40GiB    elapsed=17m9.607s   eta=1h54m50.495s
INFO [12-05|14:06:12.421] Resuming state snapshot generation       root=45703b..ab7d77 at=2147a2..323221 accounts=6,506,874 slots=39,386,076 storage=3.40GiB    elapsed=17m9.608s   eta=1h54m50.502s
INFO [12-05|14:06:12.426] Imported new chain segment               blocks=1    txs=73    mgas=22.378  elapsed=940.270ms   mgasps=23.799  number=22,157,249 hash=f7abf8..821352 dirty=230.53MiB
INFO [12-05|14:06:12.428] Unindexed transactions                   blocks=1    txs=185   tail=19,807,250 elapsed=1.596ms
INFO [12-05|14:06:20.423] Generating state snapshot                root=45703b..ab7d77 at=21b520..3d7356 accounts=6,590,363 slots=39,462,659 storage=3.41GiB    elapsed=17m17.610s  eta=1h54m2.778s
INFO [12-05|14:06:28.430] Generating state snapshot                root=45703b..ab7d77 at=220512..95ab2b accounts=6,651,700 slots=39,679,127 storage=3.43GiB    elapsed=17m25.617s  eta=1h53m42.684s
INFO [12-05|14:06:30.602] Aborting state snapshot generation       root=45703b..ab7d77 at=222552..2eb68f accounts=6,676,185 slots=39,706,103 storage=3.43GiB    elapsed=17m27.788s  eta=1h53m27.76s
INFO [12-05|14:06:30.604] Resuming state snapshot generation       root=dcfb08..ec6f79 at=222552..2eb68f accounts=6,676,185 slots=39,706,103 storage=3.43GiB    elapsed=17m27.790s  eta=1h53m27.773s
INFO [12-05|14:06:30.611] Imported new chain segment               blocks=1    txs=147   mgas=22.361  elapsed=775.910ms   mgasps=28.819  number=22,157,250 hash=c2cfb0..02b971 dirty=231.02MiB
INFO [12-05|14:06:30.612] Unindexed transactions                   blocks=1    txs=174   tail=19,807,251 elapsed=1.502ms
```

**Comment by masayil** [comment]
That's the way to confirm if your node is on correct fork.
curl -X POST -H "Content-Type: application/json" --data '{"jsonrpc":"2.0","method":"eth_getBlockByNumber","params":["0x1521600", false],"id":1}' localhost:8545 | jq ".result.hash"
The hash should be: 0x33f08e82cd658d933aec2cf90cf10d7443043c036ac4da030fc5319eba267601

**Comment by masayil** [comment]
> > > same issue
> > 
> > 
> > https://forum.polygon.technology/t/v0-2-12-beta3-upgrade-for-bor-mainnet/360 Pls check this website
> 
> After both node update to 0.2.12-beta3 and debug.setHead back to block 22156400 which about 200 blocks before hard fork, the node sync for a while. But now both the node sync stucks at block 22157055 with `Error: invalid merkle root ` like below
> 
> ```
> Error: invalid merkle root (remote: 6285eeb75ee4e86374a83a21d6c5dc0dc78db4edde8d24f3adfe690b7dd2a2ef local: fda9a13d76662405b23e45d9280e063c11b7b2bb74455bd2b49e1cdfab86c43c)
> ##############################
> 
> WARN [12-05|12:01:22.384] Synchronisation failed, dropping peer    peer=22fb22dc72570abb4faa7217319e8fed3c3183a4b46cb949c003c3adb3775585 err="retrieved hash chain is invalid: invalid merkle root (remote: 6285eeb75ee4e86374a83a21d6c5dc0dc78db4edde8d24f3adfe690b7dd2a2ef local: fda9a13d76662405b23e45d9280e063c11b7b2bb74455bd2b49e1cdfab86c43c)"
> ARN [12-05|12:01:22.385] Synchronisation failed, retrying         err="peer is unknown or unhealthy"
> INFO [12-05|12:01:22.419] Generating state snapshot                root=88aa67..7d0429 in=4567f3..406c48 at=f5cb98..a01998 accounts=13,549,678 slots=129,256,864 storage=10.52GiB   elapsed=48m35.238s  eta=2h10m37.44s
> INFO [12-05|12:01:30.427] Generating state snapshot                root=88aa67..7d0429 in=457d29..42876f at=1bb130..c1b93a accounts=13,566,105 slots=129,725,202 storage=10.55GiB   elapsed=48m43.245s  eta=2h10m46.111s
> INFO [12-05|12:01:38.435] Generating state snapshot                root=88aa67..7d0429 in=457f51..c11949 at=4a0581..ecfb3e accounts=13,567,761 slots=130,310,538 storage=10.60GiB   elapsed=48m51.253s  eta=2h11m6.297s
> ERROR[12-05|12:01:43.750] Error in block freeze operation          err="block receipts missing, can't freeze block 18450232"
> ```

I have 8 nodes .Only one node has this problem after  update to 0.2.12-beta3 and debug.setHead back to block 22155660

**Comment by ecastelli** [comment]
Same problem here.  Getting stuck on block #22158492 and then getting the following error message:
"Error: invalid merkle root (remote: 6285eeb75ee4e86374a83a21d6c5dc0dc78db4edde8d24f3adfe690b7dd2a2ef local: fda9a13d76662405b23e45d9280e063c11b7b2bb74455bd2b49e1cdfab86c43c)
##############################"

**Comment by ecastelli** [comment]
Downloading latest bor snapshot from https://matic-blockchain-snapshots.s3-accelerate.amazonaws.com/matic-mainnet/bor-pruned-snapshot-2021-12-05.tar.gz to see if this resolves the issue.

**Comment by kaber2** [comment]
> Downloading latest bor snapshot from https://matic-blockchain-snapshots.s3-accelerate.amazonaws.com/matic-mainnet/bor-pruned-snapshot-2021-12-05.tar.gz to see if this resolves the issue.

Tried on 5 nodes, no success.

**Comment by yoongyy** [comment]
All peers are dead, no one can sync now...

**Comment by lcgogo** [comment]
@yoongyy @vae520283995 @kaber2 @ecastelli 

Follow this guide to fix
https://polygontechnology.notion.site/v0-2-12-beta3-Bor-Upgrade-6c94ac47a5ac41b88bb5f3bf72cdb949

**Comment by tsutsu** [comment]
I was pulling my hair out with this, but I finally figured out what was wrong in my case.

I'm not sure how, but I somehow ended up with a copy of `genesis.json` that had a `blockAlloc` height for the new system contract of `22275000`, rather than `22156660`. So the node was hitting the hard-fork point _before_ the migration introducing the system contract was kicking in.

When I went back and re-downloaded (from https://raw.githubusercontent.com/maticnetwork/launch/master/mainnet-v1/sentry/sentry/bor/genesis.json), I got one with the right height this time, `22156660`.

For anyone having trouble, you should check that your own `genesis.json` has the proper key `"22156660"` within `blockAlloc`.

You can then verify that the migration introducing the system contract is actually being run—it will print the following line to the log when it does:

```
INFO [12-06|03:09:06.654] change contract code                     address=0x0000000000000000000000000000000000001010
```

If your sync is failing on block `22156660`, and your `bor` logs don't contain the above "change contract code" line, then you might be hitting this problem.

**Comment by joslee7410** [comment]
Good day guys,

I'm facing same problem as well.

But I already fixed it. I'm here to sharing my way but i'm not guarantee its work on everyone.

Stop the bor service, and upgrade bor to v0.2.12-beta3, which can refer by this [Link](https://forum.polygon.technology/t/v0-2-12-beta3-upgrade-for-bor-mainnet/360)

After that download the latest "genesis.json" file and "init" to your datadir.

After that start your bor service again, and right now can use bor attach to your local bor rpc

`bor attach http://127.0.0.1:8545`

and run with `debug.setHead('0x152118C')` and wait for it sync.

For my opinion, i think the critical part is you need to download the latest "genesis.json" and "init" it

Hope it can help you all. And thanks for @masayil sharing.

**Comment by wxf-kine** [comment]
> Good day guys,
> 
> I'm facing same problem as well.
> 
> But I already fixed it. I'm here to sharing my way but i'm not guarantee its work on everyone.
> 
> Stop the bor service, and upgrade bor to v0.2.12-beta3, which can refer by this [Link](https://forum.polygon.technology/t/v0-2-12-beta3-upgrade-for-bor-mainnet/360)
> 
> After that download the latest "genesis.json" file and "init" to your datadir.
> 
> After that start your bor service again, and right now can use bor attach to your local bor rpc
> 
> `bor attach http://127.0.0.1:8545`
> 
> and run with `debug.setHead('0x152118C')` and wait for it sync.
> 
> For my opinion, i think the critical part is you need to download the latest "genesis.json" and "init" it
> 
> Hope it can help you all. And thanks for @masayil sharing.




> Good day guys,
> 
> I'm facing same problem as well.
> 
> But I already fixed it. I'm here to sharing my way but i'm not guarantee its work on everyone.
> 
> Stop the bor service, and upgrade bor to v0.2.12-beta3, which can refer by this [Link](https://forum.polygon.technology/t/v0-2-12-beta3-upgrade-for-bor-mainnet/360)
> 
> After that download the latest "genesis.json" file and "init" to your datadir.
> 
> After that start your bor service again, and right now can use bor attach to your local bor rpc
> 
> `bor attach http://127.0.0.1:8545`
> 
> and run with `debug.setHead('0x152118C')` and wait for it sync.
> 
> For my opinion, i think the critical part is you need to download the latest "genesis.json" and "init" it
> 
> Hope it can help you all. And thanks for @masayil sharing.

I have the same problem and i did that like you said...but it does not work for me

**Comment by joslee7410** [comment]
> > Good day guys,
> > I'm facing same problem as well.
> > But I already fixed it. I'm here to sharing my way but i'm not guarantee its work on everyone.
> > Stop the bor service, and upgrade bor to v0.2.12-beta3, which can refer by this [Link](https://forum.polygon.technology/t/v0-2-12-beta3-upgrade-for-bor-mainnet/360)
> > After that download the latest "genesis.json" file and "init" to your datadir.
> > After that start your bor service again, and right now can use bor attach to your local bor rpc
> > `bor attach http://127.0.0.1:8545`
> > and run with `debug.setHead('0x152118C')` and wait for it sync.
> > For my opinion, i think the critical part is you need to download the latest "genesis.json" and "init" it
> > Hope it can help you all. And thanks for @masayil sharing.
> 
> > Good day guys,
> > I'm facing same problem as well.
> > But I already fixed it. I'm here to sharing my way but i'm not guarantee its work on everyone.
> > Stop the bor service, and upgrade bor to v0.2.12-beta3, which can refer by this [Link](https://forum.polygon.technology/t/v0-2-12-beta3-upgrade-for-bor-mainnet/360)
> > After that download the latest "genesis.json" file and "init" to your datadir.
> > After that start your bor service again, and right now can use bor attach to your local bor rpc
> > `bor attach http://127.0.0.1:8545`
> > and run with `debug.setHead('0x152118C')` and wait for it sync.
> > For my opinion, i think the critical part is you need to download the latest "genesis.json" and "init" it
> > Hope it can help you all. And thanks for @masayil sharing.
> 
> I have the same problem and i did that like you said...but it does not work for me

Can you try check your bor version? is it 0.2.12-stable, and make sure you running by using this "bor" 

and check your "genesis.json" file is it having this part (line 33 to 36)

```
      "blockAlloc": {
        "22156660": {
          "0000000000000000000000000000000000001010": {
            "balance": "0x0",
```

and must "init" it one more time to your datadir.

**Comment by wxf-kine** [comment]
Yes,i did everything just like you wrote above...
![image](https://user-images.githubusercontent.com/93694051/144804718-3c532292-529e-4434-9cf0-5886ec2f1ee9.png)
![image](https://user-images.githubusercontent.com/93694051/144804830-321d5f91-acf6-4026-968e-ddc9a821af1d.png)

**Comment by wxf-kine** [comment]
> > > Good day guys,
> > > I'm facing same problem as well.
> > > But I already fixed it. I'm here to sharing my way but i'm not guarantee its work on everyone.
> > > Stop the bor service, and upgrade bor to v0.2.12-beta3, which can refer by this [Link](https://forum.polygon.technology/t/v0-2-12-beta3-upgrade-for-bor-mainnet/360)
> > > After that download the latest "genesis.json" file and "init" to your datadir.
> > > After that start your bor service again, and right now can use bor attach to your local bor rpc
> > > `bor attach http://127.0.0.1:8545`
> > > and run with `debug.setHead('0x152118C')` and wait for it sync.
> > > For my opinion, i think the critical part is you need to download the latest "genesis.json" and "init" it
> > > Hope it can help you all. And thanks for @masayil sharing.
> > 
> > 
> > > Good day guys,
> > > I'm facing same problem as well.
> > > But I already fixed it. I'm here to sharing my way but i'm not guarantee its work on everyone.
> > > Stop the bor service, and upgrade bor to v0.2.12-beta3, which can refer by this [Link](https://forum.polygon.technology/t/v0-2-12-beta3-upgrade-for-bor-mainnet/360)
> > > After that download the latest "genesis.json" file and "init" to your datadir.
> > > After that start your bor service again, and right now can use bor attach to your local bor rpc
> > > `bor attach http://127.0.0.1:8545`
> > > and run with `debug.setHead('0x152118C')` and wait for it sync.
> > > For my opinion, i think the critical part is you need to download the latest "genesis.json" and "init" it
> > > Hope it can help you all. And thanks for @masayil sharing.
> > 
> > 
> > I have the same problem and i did that like you said...but it does not work for me
> 
> Can you try check your bor version? is it 0.2.12-stable, and make sure you running by using this "bor"
> 
> and check your "genesis.json" file is it having this part (line 33 to 36)
> 
> ```
>       "blockAlloc": {
>         "22156660": {
>           "0000000000000000000000000000000000001010": {
>             "balance": "0x0",
> ```
> 
> and must "init" it one more time to your datadir.

Yes  i did everything you wrote above .but when i rewind the block and synced to 22156800 ,ii still get the wrong hash
![image](https://user-images.githubusercontent.com/93694051/144804916-bd773663-bedd-41dc-9fd7-e14826899faa.png)
![image](https://user-images.githubusercontent.com/93694051/144804947-2082cb21-c431-484f-9672-78ff2fa4ecd3.png)

**Comment by joslee7410** [comment]
> > > > Good day guys,
> > > > I'm facing same problem as well.
> > > > But I already fixed it. I'm here to sharing my way but i'm not guarantee its work on everyone.
> > > > Stop the bor service, and upgrade bor to v0.2.12-beta3, which can refer by this [Link](https://forum.polygon.technology/t/v0-2-12-beta3-upgrade-for-bor-mainnet/360)
> > > > After that download the latest "genesis.json" file and "init" to your datadir.
> > > > After that start your bor service again, and right now can use bor attach to your local bor rpc
> > > > `bor attach http://127.0.0.1:8545`
> > > > and run with `debug.setHead('0x152118C')` and wait for it sync.
> > > > For my opinion, i think the critical part is you need to download the latest "genesis.json" and "init" it
> > > > Hope it can help you all. And thanks for @masayil sharing.
> > > 
> > > 
> > > > Good day guys,
> > > > I'm facing same problem as well.
> > > > But I already fixed it. I'm here to sharing my way but i'm not guarantee its work on everyone.
> > > > Stop the bor service, and upgrade bor to v0.2.12-beta3, which can refer by this [Link](https://forum.polygon.technology/t/v0-2-12-beta3-upgrade-for-bor-mainnet/360)
> > > > After that download the latest "genesis.json" file and "init" to your datadir.
> > > > After that start your bor service again, and right now can use bor attach to your local bor rpc
> > > > `bor attach http://127.0.0.1:8545`
> > > > and run with `debug.setHead('0x152118C')` and wait for it sync.
> > > > For my opinion, i think the critical part is you need to download the latest "genesis.json" and "init" it
> > > > Hope it can help you all. And thanks for @masayil sharing.
> > > 
> > > 
> > > I have the same problem and i did that like you said...but it does not work for me
> > 
> > 
> > Can you try check your bor version? is it 0.2.12-stable, and make sure you running by using this "bor"
> > and check your "genesis.json" file is it having this part (line 33 to 36)
> > ```
> >       "blockAlloc": {
> >         "22156660": {
> >           "0000000000000000000000000000000000001010": {
> >             "balance": "0x0",
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > and must "init" it one more time to your datadir.
> 
> Yes i did everything you wrote above .but when i rewind the block and synced to 22156800 ,ii still get the wrong hash ![image](https://user-images.githubusercontent.com/93694051/144804916-bd773663-bedd-41dc-9fd7-e14826899faa.png) ![image](https://user-images.githubusercontent.com/93694051/144804947-2082cb21-c431-484f-9672-78ff2fa4ecd3.png)

I running with 2 nodes, one is using the way I mention, and another node is rerun with snapshot.

Maybe you can try download the latest snapshot and extract to "bor dir"/data/bor/chaindata 

Try to redo it might solve your problem.

**Comment by wxf-kine** [comment]
> > > > > Good day guys,
> > > > > I'm facing same problem as well.
> > > > > But I already fixed it. I'm here to sharing my way but i'm not guarantee its work on everyone.
> > > > > Stop the bor service, and upgrade bor to v0.2.12-beta3, which can refer by this [Link](https://forum.polygon.technology/t/v0-2-12-beta3-upgrade-for-bor-mainnet/360)
> > > > > After that download the latest "genesis.json" file and "init" to your datadir.
> > > > > After that start your bor service again, and right now can use bor attach to your local bor rpc
> > > > > `bor attach http://127.0.0.1:8545`
> > > > > and run with `debug.setHead('0x152118C')` and wait for it sync.
> > > > > For my opinion, i think the critical part is you need to download the latest "genesis.json" and "init" it
> > > > > Hope it can help you all. And thanks for @masayil sharing.
> > > > 
> > > > 
> > > > > Good day guys,
> > > > > I'm facing same problem as well.
> > > > > But I already fixed it. I'm here to sharing my way but i'm not guarantee its work on everyone.
> > > > > Stop the bor service, and upgrade bor to v0.2.12-beta3, which can refer by this [Link](https://forum.polygon.technology/t/v0-2-12-beta3-upgrade-for-bor-mainnet/360)
> > > > > After that download the latest "genesis.json" file and "init" to your datadir.
> > > > > After that start your bor service again, and right now can use bor attach to your local bor rpc
> > > > > `bor attach http://127.0.0.1:8545`
> > > > > and run with `debug.setHead('0x152118C')` and wait for it sync.
> > > > > For my opinion, i think the critical part is you need to download the latest "genesis.json" and "init" it
> > > > > Hope it can help you all. And thanks for @masayil sharing.
> > > > 
> > > > 
> > > > I have the same problem and i did that like you said...but it does not work for me
> > > 
> > > 
> > > Can you try check your bor version? is it 0.2.12-stable, and make sure you running by using this "bor"
> > > and check your "genesis.json" file is it having this part (line 33 to 36)
> > > ```
> > >       "blockAlloc": {
> > >         "22156660": {
> > >           "0000000000000000000000000000000000001010": {
> > >             "balance": "0x0",
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > and must "init" it one more time to your datadir.
> > 
> > 
> > Yes i did everything you wrote above .but when i rewind the block and synced to 22156800 ,ii still get the wrong hash ![image](https://user-images.githubusercontent.com/93694051/144804916-bd773663-bedd-41dc-9fd7-e14826899faa.png) ![image](https://user-images.githubusercontent.com/93694051/144804947-2082cb21-c431-484f-9672-78ff2fa4ecd3.png)
> 
> I running with 2 nodes, one is using the way I mention, and another node is rerun with snapshot.
> 
> Maybe you can try download the latest snapshot and extract to "bor dir"/data/bor/chaindata
> 
> Try to redo it might solve your problem.

ok
you will use purned snapshot ? i do not know if purned snapshot  can be used...

**Comment by joslee7410** [comment]
> > > > > > Good day guys,
> > > > > > I'm facing same problem as well.
> > > > > > But I already fixed it. I'm here to sharing my way but i'm not guarantee its work on everyone.
> > > > > > Stop the bor service, and upgrade bor to v0.2.12-beta3, which can refer by this [Link](https://forum.polygon.technology/t/v0-2-12-beta3-upgrade-for-bor-mainnet/360)
> > > > > > After that download the latest "genesis.json" file and "init" to your datadir.
> > > > > > After that start your bor service again, and right now can use bor attach to your local bor rpc
> > > > > > `bor attach http://127.0.0.1:8545`
> > > > > > and run with `debug.setHead('0x152118C')` and wait for it sync.
> > > > > > For my opinion, i think the critical part is you need to download the latest "genesis.json" and "init" it
> > > > > > Hope it can help you all. And thanks for @masayil sharing.
> > > > > 
> > > > > 
> > > > > > Good day guys,
> > > > > > I'm facing same problem as well.
> > > > > > But I already fixed it. I'm here to sharing my way but i'm not guarantee its work on everyone.
> > > > > > Stop the bor service, and upgrade bor to v0.2.12-beta3, which can refer by this [Link](https://forum.polygon.technology/t/v0-2-12-beta3-upgrade-for-bor-mainnet/360)
> > > > > > After that download the latest "genesis.json" file and "init" to your datadir.
> > > > > > After that start your bor service again, and right now can use bor attach to your local bor rpc
> > > > > > `bor attach http://127.0.0.1:8545`
> > > > > > and run with `debug.setHead('0x152118C')` and wait for it sync.
> > > > > > For my opinion, i think the critical part is you need to download the latest "genesis.json" and "init" it
> > > > > > Hope it can help you all. And thanks for @masayil sharing.
> > > > > 
> > > > > 
> > > > > I have the same problem and i did that like you said...but it does not work for me
> > > > 
> > > > 
> > > > Can you try check your bor version? is it 0.2.12-stable, and make sure you running by using this "bor"
> > > > and check your "genesis.json" file is it having this part (line 33 to 36)
> > > > ```
> > > >       "blockAlloc": {
> > > >         "22156660": {
> > > >           "0000000000000000000000000000000000001010": {
> > > >             "balance": "0x0",
> > > > ```
> > > > 
> > > > 
> > > >     
> > > >       
> > > >     
> > > > 
> > > >       
> > > >     
> > > > 
> > > >     
> > > >   
> > > > and must "init" it one more time to your datadir.
> > > 
> > > 
> > > Yes i did everything you wrote above .but when i rewind the block and synced to 22156800 ,ii still get the wrong hash ![image](https://user-images.githubusercontent.com/93694051/144804916-bd773663-bedd-41dc-9fd7-e14826899faa.png) ![image](https://user-images.githubusercontent.com/93694051/144804947-2082cb21-c431-484f-9672-78ff2fa4ecd3.png)
> > 
> > 
> > I running with 2 nodes, one is using the way I mention, and another node is rerun with snapshot.
> > Maybe you can try download the latest snapshot and extract to "bor dir"/data/bor/chaindata
> > Try to redo it might solve your problem.
> 
> ok you will use purned snapshot ? i do not know if purned snapshot can be used...

Yeap, i'm using pruned snapshot. Its workable.

**Comment by ecastelli** [comment]
> Downloading latest bor snapshot from https://matic-blockchain-snapshots.s3-accelerate.amazonaws.com/matic-mainnet/bor-pruned-snapshot-2021-12-05.tar.gz to see if this resolves the issue.

Latest snapshot did not work for me.  Same error.

**Comment by wxf-kine** [comment]
> > Downloading latest bor snapshot from https://matic-blockchain-snapshots.s3-accelerate.amazonaws.com/matic-mainnet/bor-pruned-snapshot-2021-12-05.tar.gz to see if this resolves the issue.
> 
> Latest snapshot did not work for me. Same error.

It works fo me and the node is running normal now

**Comment by ecastelli** [comment]
> 

This did end up working for me as well, but I had to re-init using the correct genesis file and dump the prior chaindata first.  Thanks to everyone for providing guidance on this issue!

**Comment by ecastelli** [comment]
> > > Downloading latest bor snapshot from https://matic-blockchain-snapshots.s3-accelerate.amazonaws.com/matic-mainnet/bor-pruned-snapshot-2021-12-05.tar.gz to see if this resolves the issue.
> > 
> > 
> > Latest snapshot did not work for me. Same error.
> 
> It works fo me and the node is running normal now

This ended up working for me as well.  I just had to re-init using the correct genesis file and dump the old chainlink data first. Thanks for everyone who provided guidance!

**Comment by ruXlab** [comment]
To add to @tsutsu comment which resolved my problem

I'm running bor in the docker using [set of these docker-compose](https://github.com/vlddm/polygon-node). First I tried to update the version and use `debug.setHead()` - that didn't help. I also redownloaded snapshot but no luck

The problem was indeed related to the old `genesis.json`. Docker simply cached genesis.cache in the **builder image** and did not refresh it even with `docker compose` `--build` flag.

So to resolve the problem I got into the docker container and replaced `genesis.json`
```
host $ docker exec -it bor /bin/sh
container $ curl -s -o /genesis.json https://raw.githubusercontent.com/maticnetwork/launch/master/mainnet-v1/without-sentry/bor/genesis.json
```
Then restarted bor. Also I removed chaindata while experimenting so I had to download the recent pruned snapshot
